diff --git a/include/uxr/agent/middleware/Middleware.hpp b/include/uxr/agent/middleware/Middleware.hpp
index d3b8978..2f35d7d 100644
--- a/include/uxr/agent/middleware/Middleware.hpp
+++ b/include/uxr/agent/middleware/Middleware.hpp
@@ -26,7 +26,7 @@
 #include <chrono>
 
 #define UXR_CLIENT_DOMAIN_ID_TO_USE_FROM_REF (255)
-#define UXR_CLIENT_DOMAIN_ID_TO_OVERRIDE_WITH_ENV (255)
+#define UXR_CLIENT_DOMAIN_ID_TO_OVERRIDE_WITH_ENV (0)
 
 namespace eprosima {
 namespace uxr {
